site/content/en/docs/Integration Patterns/high-density-gameservers.md

```diff
@@ -0,0 +1,80 @@
+---
+title: "High Density GameServers"
+linkTitle: "High Density GameServers"
+date: 2021-08-31
+weight: 70
+description: >
+  How to run multiple concurrent game sessions in a single GameServer process.
+---
+
+{{< alpha title="Allocation Player Filter and Allocation State Filter" gate="PlayerAllocationFilter,StateAllocationFilter" >}}
+
+Depending on the setup and resource requirements of your game server process, sometimes it can be a more economical 
+use of resources to run multiple concurrent game sessions from within a single `GameServer` instance.
+
+The tradeoff here is that this requires more management on behalf of the integrated game server process and external 
+systems, since it works around the common Kubernetes and/or Agones container lifecycle.
+
+Utilising the new allocation `gameServerState` filter as well as the existing ability to edit the 
+`GameServer` Labels at both [allocation time]({{% ref "/docs/Reference/gameserverallocation.md" %}}), and from 
+within the game server process, [via the SDK]({{% ref "/docs/Guides/Client SDKs/_index.md#setlabelkey-value" %}}), 
+means Agones is able to atomically remove a `GameServer` from the list of potentially allocatable 
+`GameServers` at allocation time, and then return it back into the pool of allocatable `GameServers` if and when the 
+game server process deems that is has room to host another game session. 
+
+<a href="../../../diagrams/high-density.puml.png" target="_blank">
+<img src="../../../diagrams/high-density.puml.png" alt="High Density Allocation Diagram" />
+</a>
+
+{{< alert title="Info" color="info">}}
+To watch for Allocation events, there is the initial `GameServer.status.state` change from `Ready` to `Allocated`,
+but it is also useful to know that the value of `GameServer.metadata.annotations["agones.dev/last-allocated"]` will
+change as it is set by Agones with each allocation with the current timestamp, regardless of if there 
+is a state change or not.
+{{< /alert >}}
+
+## Example `GameServerAllocation`
+
+The below `Allocation` will first attempt to find a `GameServer` from the `Fleet` `simple-udp` that is already 
+Allocated and also has the label `agones.dev/sdk-available` with the value of `true`.
+
+The above condition indicates that the matching game server process behind the matched `GameServer` record is able to 
+accept another game session at this time.
+
+If the above Allocated `GameServer` does not exist, then allocated a Ready `GameServer` from the `simple-udp` `Fleet`.
```

Comment:
nit: what about

If an Allocated `GameServer` does not exist with the desired labels, then use the next selector to allocate a Ready `GameServer` from the `simple-udp` `Fleet`.

Reply:
I like it!